### PR TITLE
Reduce code complexity

### PR DESCRIPTION
**Pull request by vehre**
This pull request removes code complexity and improves readability by replacing conditional compilation blocks by macro calls. The macros contain exactly the same code like the conditional compilation blocks, but are far easier to read and understand.

This pull request replaces #300.

### CODE REVIEW

**Comment by zbeekman** [comment]
@vehre: strangely both the pull request CI test and the push CI test both errored on OS X, only with code coverage flags, for np=7 ring abort test. On one test image 2 raised the abort, on the other image `num_images()` raised the abort. (I know you can't look at this until Monday, but I just wanted to add this observation here, for later)

File: src/mpi/mpi_caf.c [hunk 1]
```diff
@@ -3715,37 +3605,31 @@ PREFIX (atomic_op) (int op, caf_token_t token ,
...
+      ierr = MPI_Fetch_and_op(value, old, dt, image, offset, MPI_BOR, *p);
+      break;
+    case 5:
+      ierr = MPI_Fetch_and_op(value, old, dt, image, offset, MPI_BXOR, *p);
+      break;
+    default:
+      printf ("We apologize but the atomic operation requested for MPI < 3 is not yet implemented\n");
+      break;
+    }
+  CAF_Win_unlock (image, *p);
```
Comment (zbeekman) [inline src/mpi/mpi_caf.c]: @vehre Did anything other than indentation change here? Are you comfortable amending commits? If so could you changed the indentation back? It's hard to see what has changed when there are trivial indentation changes introduced.

**Comment by zbeekman** [comment]
LGTM

File: src/mpi/mpi_caf.c [hunk 2]
```diff
@@ -3715,37 +3605,31 @@ PREFIX (atomic_op) (int op, caf_token_t token ,
...
+    case 4:
+      ierr = MPI_Fetch_and_op(value, old, dt, image, offset, MPI_BOR, *p);
+      break;
+    case 5:
+      ierr = MPI_Fetch_and_op(value, old, dt, image, offset, MPI_BXOR, *p);
+      break;
+    default:
+      printf ("We apologize but the atomic operation requested for MPI < 3 is not yet implemented\n");
+      break;
+    }
```
Comment (zbeekman) [inline src/mpi/mpi_caf.c]: @vehre Did anything other than indentation change here? Are you comfortable amending commits? If so could you changed the indentation back? It's hard to see what has changed when there are trivial indentation changes introduced.

**Comment by zbeekman** [comment]
Let's see if re-running tests causes them to pass now...

**Comment by zbeekman** [comment]
Ring abort is still failing, on OS and Linux intermittently... I'm tempted to merge this PR anyway, even though this test is failing, since the failing test is irrelevant to the changes in the PR...

**Comment by zbeekman** [comment]
I've rebased this branch onto vehre/sync_images_fix_new so that we can see that the tests pass...

File: src/mpi/mpi_caf.c (hunk 2 above)
Comment (vehre) [inline src/mpi/mpi_caf.c]: Only indentation changed. The conditional compilations above drove the indenter nuts and by getting rid of the preprocessor directives indentation went fine. So I included it into the commit.

**Comment by vehre** [comment]
The test failures should have the same source as pointed out in #313 .

**Comment by zbeekman** [comment]
LGTM

File: src/mpi/mpi_caf.c (hunk 2 above)
Comment (zbeekman) [inline src/mpi/mpi_caf.c]: :+1:

**Pull request merged**